src/java.base/share/classes/java/io/Reader.java

```diff
@@ -198,6 +202,8 @@ public int read(CharBuffer target) throws IOException {
         } else {
             int len = target.remaining();
             char[] cbuf = new char[len];
+            // If a read-only check had not been done above, then
+            // the stream would be incorrectly advanced here.
```

Comment:
The RO check is good but I'm not sure that this comment is useful (it might confuse future maintainers), I would be tempted to leave that part out.